### PR TITLE
[TR] Fix archive to download when exporting items with media

### PR DESCRIPTION
**Pull request by willy-ahva**
This PR fixes a bug with CSV Product Export, XLSX Product Export, but CSV variant group Export is ok. When we export items with media, there is no archive to download with media.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | ok
| Added Behats                      | They are in 1.4, need to be remontay Waiting for CI
| Changelog updated                 | Bug has been merge in master, no need to update
| Review and 2 GTM                  | 
| Micro Demo to the PO (Story only) |
| Migration script                  |
| Tech Doc                          |

### CODE REVIEW

File: src/Pim/Component/Connector/Writer/File/XlsxProductWriter.php [hunk 1]
```diff
@@ -13,7 +12,7 @@
  * @copyright 2016 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
-class XlsxProductWriter extends AbstractFileWriter implements ItemWriterInterface
```
Comment (willy-ahva) [inline src/Pim/Component/Connector/Writer/File/XlsxProductWriter.php]: No neeed to implement `ItemWriterInterface` which is already in `AbstractFileWriter`

**Comment by aRn0D** [comment]
GTM

File: src/Pim/Component/Connector/Writer/File/XlsxProductWriter.php (hunk 1 above)
Comment (jjanvier) [inline src/Pim/Component/Connector/Writer/File/XlsxProductWriter.php]: I disagree
the behaviour of `XlsxProductWriter` is to implement `ItemWriterInterface`

what tomorrow, `AbstractFileWriter` is dropped, or `XlsxProductWriter` extends something else, the class is "broken"

so please, implement `ItemWriterInterface`

**Comment by jjanvier** [comment]
GTM once fixed

File: src/Pim/Component/Connector/Writer/File/BulkFileExporter.php [hunk 2]
```diff
@@ -46,6 +52,17 @@ public function exportAll(array $items, $target)
...
+     *      [
+     *          'message' => (string),
+     *          'medium'  => [
+     *              'filePath'     => (string),
+     *              'exportPath'   => (string),
+     *              'storageAlias' => (string)
+     *          ]
+     *      ],
+     *      [...]
+     *  ]
```
Comment (phaseinducer) [inline src/Pim/Component/Connector/Writer/File/BulkFileExporter.php]: :+1:

File: src/Pim/Component/Connector/Writer/File/BulkFileExporter.php [hunk 3]
```diff
@@ -53,40 +70,71 @@ public function getErrors()
...
             'message' => $message,
             'medium'  => $medium,
         ];
     }
+
+    /**
+     * @param array  $medium
+     * @param string $filePath
+     */
+    protected function addCopiedMedium(array $medium, $filePath)
```
Comment (phaseinducer) [inline src/Pim/Component/Connector/Writer/File/BulkFileExporter.php]: Maybe call it `$copyPath` like the array key ? `filePath` is already used as key in the $medium array, it's confusing.

File: src/Pim/Component/Connector/Writer/File/CsvProductWriter.php [hunk 4]
```diff
@@ -49,6 +49,10 @@ public function write(array $items)
 
         $this->mediaCopier->exportAll($media, $exportDirectory);
 
+        foreach ($this->mediaCopier->getCopiedMedia() as $copy) {
```
Comment (phaseinducer) [inline src/Pim/Component/Connector/Writer/File/CsvProductWriter.php]: `$media` would be more accurate imho.

File: src/Pim/Component/Connector/Writer/File/XlsxProductWriter.php [hunk 5]
```diff
@@ -48,12 +57,16 @@ public function write(array $items)
...
         foreach ($items as $item) {
             $products[] = $item['product'];
-            $media[] = $item['media'];
+            $media[]    = $item['media'];
         }
 
         $this->flatRowBuffer->write($products, $this->isWithHeader());
         $this->mediaCopier->exportAll($media, $exportDirectory);
 
+        foreach ($this->mediaCopier->getCopiedMedia() as $copy) {
```
Comment (phaseinducer) [inline src/Pim/Component/Connector/Writer/File/XlsxProductWriter.php]: idem

**Comment by phaseinducer** [comment]
Little naming comments then GTM

File: src/Pim/Component/Connector/Writer/File/CsvProductWriter.php (hunk 4 above)
Comment (willy-ahva) [inline src/Pim/Component/Connector/Writer/File/CsvProductWriter.php]: But it's not really the media. That's why I didn't named it $media. It's the path of the copied media and the original media.

File: src/Pim/Component/Connector/Writer/File/BulkFileExporter.php (hunk 3 above)
Comment (willy-ahva) [inline src/Pim/Component/Connector/Writer/File/BulkFileExporter.php]: hum yeah why not

File: src/Pim/Component/Connector/Writer/File/CsvProductWriter.php (hunk 4 above)
Comment (phaseinducer) [inline src/Pim/Component/Connector/Writer/File/CsvProductWriter.php]: Right, ok !

**Pull request merged**